### ISSUE TITLE
Charset get lost and casing of header keys changes

### ISSUE THREAD

**Issue by iongion** [smhg/gettext-parser#12]
Headers act weird:

```
        var translation = {
          charset: "UTF-8",
          headers: {
            "Project-Id-Version": "project 1.0.2",
            "Mime-Version": "1.0",
            "Content-Type": "text/plain; charset=UTF-8",
            "Content-Transfer-Encoding": "8bit",
            "Plural-Forms": "nplurals=2; plural=(n!=1);",
            "X-Poedit-SourceCharset": "UTF-8"
          },
          translations: {}
        }
```

And the generated files outputs:

```
msgid ""
msgstr ""
"Project-Id-Version: project 1.0.2\n"
"MIME-Version: 1.0\n"
"Content-Transfer-Encoding: 8bit\n"
"Plural-Forms: nplurals=2; plural=(n!=1);\n"
"X-Poedit-Sourcecharset: UTF-8\n"
"Content-Type: text/plain;\n"
```

As you can see, `Content-Type` header lacks the charset and then the `X-Poedit-SourceCharset` header casing is weird and confuses 3rd party editors, it becomes `X-Poedit-Sourcecharset`

Can't the headers be sent as they are ? Without any magic title casing ?

**Comment by smhg** [comment]
Hi! I took over this project and would like to follow up on this.
Did you find a workaround? Or it isn't breaking anything?

**Comment by FredyC** [comment]
@smhg Looks like you took over and abandoned again? :) What a pity. Just encountered this issue. Are you available at least for merging PR and publishing or did you lost interest completely?

Either way, I don't agree there should be any "uppercasing" happening. It should really be left in the same state as it was read. Besides various editors doing some shenanigans, this is also super annoying for file versioning. One tool changes headers some way, other tool does it differently and if it happens between commits, it's full of unrelated changes.

**Comment by smhg** [comment]
I do maintain this project. The few open issues are just the ones for which I'd appreciate PR's.
Sidenote: for some weird reason I don't get e-mailed when there is a new issue or PR unless you ping me with an `@username` (so please feel free to always do that).

Concerning the headers: I think we are all on the same page here. Feel free to submit your solution. Since these are 2 separate issues, maybe separate them into 2 PR's?

**Comment by smhg** [comment]
Care to share what worked for you so others can benefit from it?

**Comment by FredyC** [comment]
https://github.com/rubenv/pofile

**Comment by smhg** [comment]
I've changed how headers are handled.
The ones from [this list](https://www.gnu.org/software/gettext/manual/html_node/Header-Entry.html#Header-Entry) have a fixed title case format. All others are left unchanged.

**Comment by smhg** [comment]
Released on npm as `4.0.0-alpha.0`.
@iongion @FredyC @vallo87 @AlexMost it would be great if one of you could give it a try.

**Comment by AlexMost** [comment]
I have checked this on my use case and seems like everything is ok! Thanks a lot!
